MenuBar related classes and TreeEvent class emulated.

nihavend opened the pull request:
@aharui TreeEvent and the menubar related classes are emulated.

aharui commented:
We definitely want you to stop and consider whether any class is absolutely required before adding it.  What code is accessing MenuBarItem and why?
Also, we are not planning to support embedded fonts for a while, if ever, so we don't want to support IFontContextComponent APIs at this time.

aharui commented on frameworks/projects/MXRoyale/src/main/royale/mx/controls/listClasses/IListItemRenderer.as:
```diff
@@ -0,0 +1,85 @@
+package mx.controls.listClasses
+{
+	COMPILE::SWF {
+		import flash.events.IEventDispatcher;
```
This doesn't seem right.  I think it can be org.apache.royale.events.IEventDispatcher even for SWF

nihavend replied:
I looked at the previous samples to do for the older flash classes and used the same way Alex. If it is not the right way, ok i am going to change it.

nihavend commented:
@aharui, believe me that I am.not med to work for hours for nights to code for junk.

menuitem is required because I am using menubar and you can check my codebase if you want.

I have deleted lots of work according to your advises but they were not fully unnecessary codes.

They were required by the ones you advise me not to work on for just now and commented out for later and because i have already emulated the dependents, i added them to git. And deleted later.

When i begin to emulate a class, i try to go to deeper where possible. I am not aware or have knowledge what will be planned and not to be covered by royale.

My first rule to stop where i can not go deeper because of dependency and to covert simple classes like IFontContextComponent that has few lines and has no dependency to un-emulated complex classes.

After emulating components,

1. i am building the changes over nightly build first (mx and spark)
2. copy them to git repo and build there  (mx and spark)
3. I am checking the build for my project with the newly generated libararies
4. pushing to github

So i all my emulations must be compiled successfully before pushed to github.

I am not happy with this because i am working for nothing but junk. I am trying to help you but i see that it is not the right way.

Okay let us stop here because i see that we both do not have to spend our valuable time for nothing. It looks like it is waste of time.

So  may you offer a new way working so i may go on working on the project.

aharui commented:
Serkan,

I'm sure it is difficult and frustrating for you because you do not have strong knowledge of the Flex internal code.

Because there isn't a large team working on the emulation, and you and Alina and Pashmina and others do not have strong knowledge of Flex, I think the right goal is to bring over as little Flex code as possible.  There are a couple of exceptions like RemoteObject and Charts, but fundamentally, the less code we have to get working, the sooner your application will run, and it will download faster too.

So, when you get a compile error that you need some class, I think the first question is whether you really need it or not.  I just scanned your code for MenuBarItem and I only found an import, but no actual usage.  Sometimes imports are left in by accident.  I've found several when porting the Chart code.   But also, MenuBarItem is an "internal" class so that's why I wanted to know why you needed it.  Most of the time I think you can use MenuBar without referencing MenuBarItem.

Other things to consider are how something would be implemented without Flash.  And performance and download size.

Regarding a more efficient workflow, it sounds like you can build the code in the repos, so I would skip the nightly builds and just sync up the 3 repos and build them.  Then build your project.  That's how I'm working.  I never bother with the nightly builds.

Also, if you can subset your application in some way, you may find it makes it easier to get to certain milestones like compiling without errors, or showing a first screen.

If you have other suggestions on how to make the process easier or more efficient, suggest it on the dev@royale mailing list.  Unless someone comes up with another major chunk of code like Charts to migrate, once I get Charts working, I should be able to help just getting things to run better.  Alina was first to sign up so she usually gets my attention first, but you are important as well and I will generally choose the problem that most likely will be hard for folks without strong knowledge of Flex to resolve.

nihavend commented:
Alex,
I completely agree with you Alex. I am not good at Flex as Java. I learned as needed for UI.
It is my fault that i should have checked the import statements carefully.

I am removing menubaritem and related classes.

May you clarify what you mean by 3 repo ?

Currently i am having issues with my email hosting. Can not send mail. FYI.

aharui commented:
All of Royale is in 3 repos:  
-royale-asjs is the ActionScript code
-royale-compiler is the compiler source code
-royale-typedefs is externs/typedefs for 3rd party libraries

If you can follow the readme in royale-asjs, it should tell you to set up some environment variables and then run "ant all" and it should clone the royale-typedefs and royale-compiler repos and build them.  If you can do that, you won't need nightly builds.  You would just sync up the 3 repos.  I think in one of the threads I explained to Alina how to build an application from the jars and SWCs in the royale-asjs repo after all 3 have been built

nihavend commented:
Okay, i am currently just snycing only royale-asjs and building only MXRoyale(pom) and SparkRoyale(pom) folders before compiling my application.

I am going to check the threads follow your advices.

Thanks.

aharui commented:
I forgot to mention that you don't have to compile all of royale-asjs if a change is made only to MXRoyale or SparkRoyale.  You can run Ant or Maven in a folder like frameworks/projects/MXRoyale and once it completes you can go back to compiling the application.

When syncing up the repos, you should build the entire repo.  On my Mac, it takes 3 minutes to build the compiler, and 4 to 6 minutes to build royale-asjs.  If Windows is slower, you can try disabling some of the virus scanners on the repo folders.

nihavend commented:
It is what i am doing Alex. I just compile and MXRoyale and SparkRoyale with maven.

I am going to check the building of repor. 
Thanks.

aharui commented:
If you are using Maven, then you have to clone each repo individually.  Then run "mvn clean install" in royale-compiler, then the same command in royale-typedefs.  If you pick up any changes in those repos, just run mvn clean install again in that repo.

carlosrovira commented:
Hi, just to complete what Alex said about Maven. If you change something only in MXRoyale, you only  need to run "mvn clean install" in that project (don't need to do in the whole repo).

Depending on the repo I use to add "-DskipTests" to avoid test and compile even faster (but remember to run without this from time to time).

nihavend commented:
I compile both folders :
mxbuild.bat : set MAVEN_OPTS=-Duser.country=EN -Duser.language=en
(cd D:\dev\royale-emulation-works\github\royale-asjs\frameworks\projects\MXRoyale && mvn -DskipTests -Drat.skip=true clean install)

sparkbuild.bat : 

set MAVEN_OPTS=-Duser.country=EN -Duser.language=en
(cd D:\dev\royale-emulation-works\github\royale-asjs\frameworks\projects\SparkRoyale && mvn -DskipTests -Drat.skip=true clean install)

aharui commented:
IUITextField is reporting conflicts. I had to disable many APIs in IUITextField for now to make it practical to have a browser implementation of IUITextField.  Can you remove your changes and try to see if the one that is currently in the repo is sufficient for your needs?
   
Also, can you remove CSSStyleDeclaration from the PR?  Seems like it is just a whitespace change?

nihavend commented:
Classes removed.

aharui commented:
Hi Serkan,

I'm sorry I was not more careful with my words.  I didn't want you to delete IUITextField.as or CSSStyleDeclaration from the repo, I wanted you to revert the changes you made to those files so that they were no longer in the pull request.  The pull request makes it appear that you deleted the files.  I'm surprised that the build succeeded for you after you did that.  Or am I just not reading the PR correctly?

nihavend commented:
Hi Alex,




I dıd not compile my project after deleting. Because I will be away for few days, I left checking for later.




I may fix it after sunday if it is ok for you.




Android için Outlook uygulamasını edinin







On Fri, Nov 2, 2018 at 12:13 AM +0300, "aharui" <notifications@github.com> wrote:












Hi Serkan,


I'm sorry I was not more careful with my words.  I didn't want you to delete IUITextField.as or CSSStyleDeclaration from the repo, I wanted you to revert the changes you made to those files so that they were no longer in the pull request.  The pull request makes it appear that you deleted the files.  I'm surprised that the build succeeded for you after you did that.  Or am I just not reading the PR correctly?



—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.

nihavend commented:
@aharui i checked-out two files from upstream and updated PR. I hopechanges are reverted back to original repo for IUITextField.as and CSSStyleDeclaration.as.
